### ISSUE TITLE
Update/Add Visual Studio 2019

### ISSUE THREAD

**Issue by zufuliu** [actions/virtual-environments#932]
**Tool information**
- Tool name: Visual Studio 2019
- Tool license: 
- Add or update? Update
- Desired version: 16.6.0 or later
- Approximate size: 
- If this is an add request:
  - Brief description of tool: 
  - URL for tool's homepage: 

**Area for Triage**:

C/C++

**Question, Bug, or Feature?**:

Feature

**Virtual environments affected**
- [ ] macOS 10.15
- [ ] Ubuntu 16.04 LTS
- [ ] Ubuntu 18.04 LTS
- [ ] Windows Server 2016 R2
- [x] Windows Server 2019

**Can this tool be installed during the build?**

**Tool installation time in runtime**

**Are you willing to submit a PR?**

The current version Visual C++ 14.25.28610 (16.4) is outdated, but 14.26.28801 (16.6) is available. 
It cause build errors for Clang (see https://github.com/zufuliu/notepad2/runs/701136763?check_suite_focus=true#step:4:8), because of `_SILENCE_CLANG_CONCEPTS_MESSAGE` (see https://github.com/microsoft/STL/issues/105), this macro is no longer defined in lasted Visual C++ headers.

**Comment by zufuliu** [comment]
See also https://github.com/microsoft/STL/issues/480 and https://github.com/microsoft/STL/pull/708

**Comment by maxim-lobanov** [comment]
Hello!
We always install latest VS version during image-generation. VS 16.6 will be delivered next week

**Comment by zufuliu** [comment]
Thank you.

**Comment by LanceMcCarthy** [comment]
Hi Team, just to make sure, can you please double check that when updating to VS 16.6 that the new UWP SDK (version 19041) is also included.

Otherwise, you will see a bunch of Actions fail to build, like these two:

- https://github.com/microsoft/github-actions-for-desktop-apps/pull/30/checks?check_run_id=709249059 
- https://github.com/LanceMcCarthy/MediaFileManager/actions/runs/113382459

**Comment by maxim-lobanov** [comment]
Hi @LanceMcCarthy , we are planning to deliver VS 16.6 this week and new SDK (19041) next week.
Just to clarify: Do you mean that builds will fail if VS 16.6 is installed on image but 19041 SDK not?
In your builds, I see that VS 16.5 is still used.

**Comment by LanceMcCarthy** [comment]
Correct, indeed the reason it fails is because it's missing 19041. 

It's not a guarantee that you get 19041 just for installing VS 16.6 (I had to go in and explicitly add it after the fact in VS Installer > Additional Components), so I wanted to mention it now so that you wouldn't need to roll out two updates :)

**Comment by maxim-lobanov** [comment]
I see. So VS 16.6 should be delivered in a few days. And 19041 will be delivered next week, PR is ready and will be merged today (#935)

**Comment by asklar** [comment]
> Hello!
> We always install latest VS version during image-generation. VS 16.6 will be delivered next week

@maxim-lobanov 
The page https://github.com/actions/virtual-environments/blob/master/images/win/Windows2019-Readme.md

lists Version: VisualStudio/16.5.5+30104.148
However this has been updated to 16.6 already.

**Comment by maxim-lobanov** [comment]
@asklar , Yep, it is expected. Usually, deployment of all VMs take 4-5 days. We update master branch only when update is delivered to all machines.

**Comment by asklar** [comment]
@maxim-lobanov thanks.
Our team's CI automatically started using VS 16.6. We started to see our code fail to build due to a confirmed regression in the compiler. The readme on the front page says it's still using 16.5. 
The "receive updates" links on the readme just mention general github changes (i.e. feature updates), not updates to specific images like windows-2019. 

a) how can we know what changes to the windows-2019 image are coming in and when
b) is it possible to set a version of VS so that it doesn't get updated as often? we've been burned by regressions in VS's compiler/linker/packaging pretty much every release.

**Comment by asklar** [comment]
filed https://github.com/actions/virtual-environments/issues/963
